xmake/toolchains/llvm/xmake.lua

```diff
@@ -87,26 +87,37 @@ toolchain("llvm")
         if toolchain:is_plat("windows") and not is_host("windows") then
             toolchain:add("ldflags", "-fuse-ld=lld")
             toolchain:add("shflags", "-fuse-ld=lld")
-        elseif toolchain:is_plat("macosx") then
-            if not toolchain:config("xcode_sysroot") then
-                local xcode_dir     = get_config("xcode")
-                local xcode_sdkver  = toolchain:config("xcode_sdkver")
-                local xcode_sdkdir
-                if xcode_dir and xcode_sdkver then
-                    local macsdk = path.join(xcode_dir, "Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX" .. xcode_sdkver .. ".sdk")
-                    if os.isdir(macsdk) then
-                        xcode_sdkdir = macsdk
-                    end
-                else
+        elseif toolchain:is_plat("macosx", "iphoneos", "tvos", "watchos", "xros") then
+            if not toolchain:config("xcode_sysroot") and not get_config("xcode_sysroot") then
+                local xcode_dir     = toolchain:config("xcode") or get_config("xcode")
+                local xcode_sdkver  = toolchain:config("xcode_sdkver") or get_config("xcode_sdkver")
+                local sdkdir
+                if is_plat("macosx") then
                     -- @see https://github.com/xmake-io/xmake/issues/1179
-                    local macsdk = "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk"
-                    if os.isdir(macsdk) then
-                        xcode_sdkdir = macsdk
+                    local sdkroot = xcode_dir and path.join(xcode_dir, "Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs")
+                                              or "/Library/Developer/CommandLineTools/SDKs"
+                    if xcode_sdkver and os.isdir(path.join(sdkroot, "MacOSX" .. xcode_sdkver .. ".sdk")) then
+                        sdkdir = path.join(sdkroot, "MacOSX" .. xcode_sdkver .. ".sdk")
+                    elseif os.isdir(path.join(sdkroot, "MacOSX.sdk")) then
+                        sdkdir = path.join(sdkroot, "MacOSX.sdk")
                     end
+                elseif is_plat("iphoneos", "tvos", "watchos", "xros") then
+                    local mapper = {
+                        iphoneos = { arm64 = "iPhoneOS", ["x86_64"] = "iPhoneSimulator" },
```

Comment:
This is inaccurate; arm64 could also be iPhoneSimulator.

https://github.com/xmake-io/xmake/blob/4b0a1d6e4eb5ce15b86f75a3e8ca85041c445ad0/xmake/toolchains/xcode/check.lua#L106-L133

we can move these common check code to toolchain_utils

https://github.com/xmake-io/xmake/blob/dev/xmake/modules/private/utils/toolchain.lua